homeassistant/components/notify/__init__.py

```diff
@@ -32,6 +34,13 @@
 
 SERVICE_NOTIFY = "notify"
 
+NOTIFY_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_MESSAGE): cv.template,
+    vol.Optional(ATTR_TITLE, default=ATTR_TITLE_DEFAULT): cv.string,
+    vol.Optional(ATTR_TARGET): cv.string,
+    vol.Optional(ATTR_DATA): dict,      # nobody seems to be using this (yet)
```

Comment:
Just keep it as a dict, it's because of extra data that you might pass into the notification (action callbacks, badges, platform specific features)